src/tools/compiletest/src/main.rs

```diff
@@ -1,10 +1,18 @@
-use std::{env, sync::Arc};
+use std::{env, io::IsTerminal, sync::Arc};
 
 use compiletest::{common::Mode, log_config, parse_config, run_tests};
 
 fn main() {
     tracing_subscriber::fmt::init();
 
+    // colored checks stdout by default, but for some reason only stderr is a terminal.
+    // compiletest *does* print many things to stdout, but it doesn't really matter.
+    if std::io::stderr().is_terminal()
+        && matches!(std::env::var("NO_COLOR").as_deref(), Err(_) | Ok("0"))
```

Comment:
shouldn't this be specifically `Err(Not found)`? I think if you have Bob utf-8 in `NO_COLOR` you shouldn't get color

Reply:
We probably should error if we get bob utf-8, but yeah I think checking for `VarError::NotPresent` is probably less surprising.

Reply:
i was just matching what colored does and i think it *really* doesn't matter. do you really want me to open the workspace again and change it? :3

Reply:
if someone has invalid utf8 in their `NO_COLOR` variable they deserve anything but attention to handling it

Reply:
If someone runs into this, they can open an issue 🤷

Reply:
please ping me on the issue so i can close it immediately